packages/gatsby-theme-carbon/src/components/GlobalSearch/GlobalSearch.js

```diff
@@ -180,6 +180,12 @@ const GlobalSearchInput = () => {
             onClick={() => {
               toggleNavState('searchIsOpen', 'open');
               toggleNavState('switcherIsOpen', 'close');
+              document.addEventListener('keyup', function(e){ 
```

Comment:
We'll want to move this to a `useEffect` hook. Putting it here means the event listener will be added to the document every time the button is clicked.

Reply:
@aliunwala do you need some assistance with this? it's a little reactspecific

Reply:
I'll give it a try shortly and see if I can get it to work. Thanks for reaching out.